test/unit/cmap/handshake/client_metadata.test.ts

```diff
@@ -1,149 +1,523 @@
+            region: 'region'
+          }
+        },
+        {
+          context: 'FUNCTION_TIMEOUT_SEC provided',
+          env: [
+            ['FUNCTION_NAME', 'non-empty'],
+            ['FUNCTION_TIMEOUT_SEC', '12345']
+          ],
+          outcome: {
+            name: 'gcp.func',
+            timeout_sec: new Int32(12345)
+          }
+        }
+      ],
+      vercel: [
+        {
+          context: 'no additional metadata',
+          env: [['VERCEL', 'non-empty']],
+          outcome: {
+            name: 'vercel'
+          }
+        },
+        {
+          context: 'VERCEL_REGION provided',
+          env: [
+            ['VERCEL', 'non-empty'],
+            ['VERCEL_REGION', 'region']
+          ],
+          outcome: {
+            name: 'vercel',
+            region: 'region'
+          }
+        }
+      ]
+    };
+
+    for (const [provider, testsForEnv] of Object.entries(tests)) {
+      context(provider, () => {
+        for (const { context, env: faasVariables, outcome } of testsForEnv) {
+          const setupEnv = () => {
+            for (const [k, v] of faasVariables) {
+              if (v != null) {
+                process.env[k] = v;
+              }
+            }
+          };
+
+          const cleanupEnv = () => {
+            for (const [k] of faasVariables) {
+              delete process.env[k];
+            }
+          };
+
+          it(context, () => {
+            try {
+              setupEnv();
+              const { env } = makeClientMetadata({ driverInfo: {} });
+              expect(env).to.deep.equal(outcome);
+            } finally {
+              cleanupEnv();
+            }
+          });
+
+          it('always places name as the last key', () => {
+            try {
+              setupEnv();
+              const keys = Array.from(getFAASEnv()?.keys() ?? []);
+              expect(keys).to.have.property(`${keys.length - 1}`, 'name');
+            } finally {
+              cleanupEnv();
+            }
+          });
+        }
+      });
+    }
+
+    context('when a numeric FAAS env variable is not numerically parsable', () => {
+      before(() => {
+        process.env.AWS_EXECUTION_ENV = 'non-empty-string';
+        process.env.AWS_LAMBDA_FUNCTION_MEMORY_SIZE = '123not numeric';
+      });
 
-    context('when the app name is under 128 bytes', () => {
-      const options = {
-        appName: 'myApplication',
-        driverInfo: {}
-      };
-      const metadata = makeClientMetadata(options);
+      after(() => {
+        delete process.env.AWS_EXECUTION_ENV;
+        delete process.env.AWS_LAMBDA_FUNCTION_MEMORY_SIZE;
+      });
+
+      it('does not attach it to the metadata', () => {
+        expect(makeClientMetadata({ driverInfo: {} })).not.to.have.nested.property('aws.memory_mb');
+      });
+    });
+  });
+
+  describe('metadata truncation', function () {
+    afterEach(() => sinon.restore());
+
+    context('when driverInfo is too large', () => {
```

Comment:
error throwing isn't truncation, these cases should live with the respective blocks for setting this info above

Reply:
Moved up to the section on driverInfo